src/mozilla_sec_eia/models/sec_eia_record_linkage/transform_sec_input.py

```diff
@@ -0,0 +1,392 @@
+        df=basic_10k_df, key_cols=["company_name", "street_address"]
+    )
+
+    return basic_10k_df
+
+
+@asset(
+    ins={
+        "basic_10k_dfs": AssetIn("basic_10k_company_info"),
+        "sec10k_filing_metadata_dfs": AssetIn("sec10k_filing_metadata"),
+    },
+)
+def core_sec_10k__filers(
+    basic_10k_dfs: dict[str, pd.DataFrame],
+    sec10k_filing_metadata_dfs: dict[str, pd.DataFrame],
+) -> pd.DataFrame:
+    """Asset for creating a cleaned basic 10k table with EIA utility matched.
+
+    Flatten the table across time to only keep the most recent record
+    for each unique company name and address pair. Clean table and link filers
+    to EIA utilities.
+    """
+    basic_10k_df = pd.concat(basic_10k_dfs.values())
+    sec10k_filing_metadata = pd.concat(sec10k_filing_metadata_dfs.values())
+    basic_10k_df = transform_basic10k_table(basic_10k_df, sec10k_filing_metadata)
+    out_df = basic_10k_df.fillna(np.nan).reset_index(names="record_id")
+    # match EIA utilities to filers
+    # TODO: Here we conduct the match to EIA and add on a column with utility_id_eia
+    return out_df
+
+
+@asset(
+    ins={
+        "sec_10k_filers_matched_df": AssetIn("core_sec_10k__filers"),
+        "clean_ex21_df": AssetIn("transformed_ex21_subsidiary_table"),
+        "clean_eia_df": AssetIn("core_eia__parents_and_subsidiaries"),
+    },
+)
+def out_sec_10k__parents_and_subsidiaries(
+    sec_10k_filers_matched_df: pd.DataFrame,
+    clean_ex21_df: pd.DataFrame,
+    clean_eia_df: pd.DataFrame,
+) -> pd.DataFrame:
+    """Asset for creating an SEC 10K output table.
+
+    Add in Ex. 21 subsidiaries and link them to already present
+    filing companies. Create an sec_company_id for subsidiaries
+    that aren't linked to a CIK.
+    """
+    ex21_df_with_cik = match_ex21_subsidiaries_to_filer_company(
+        basic10k_df=sec_10k_filers_matched_df, ex21_df=clean_ex21_df
+    )
+    sec_10k_filers_matched_df = sec_10k_filers_matched_df.merge(
+        ex21_df_with_cik[["central_index_key", "parent_company_cik", "own_per"]],
+        how="left",
+        on="central_index_key",
+    )
+    # get the subsidiary companies that weren't matched to a 10K filing company
+    ex21_non_filing_subs_df = ex21_df_with_cik[
+        ex21_df_with_cik["central_index_key"].isnull()
+    ]
+    ex21_non_filing_subs_df.loc[:, "files_10k"] = False
+    # the last step is to take the EIA utilities that haven't been matched
+    # to a filer company, and merge them by company name onto the Ex. 21 subs
+    unmatched_eia_df = clean_eia_df[
+        ~clean_eia_df["utility_id_eia"].isin(
+            sec_10k_filers_matched_df.utility_id_eia.unique()
+        )
+    ].drop_duplicates(subset="company_name")
+    ex21_non_filing_subs_df = ex21_non_filing_subs_df.merge(
+        unmatched_eia_df[["utility_id_eia", "company_name"]],
+        how="left",
+        on="company_name",
+    ).drop_duplicates(subset="sec_company_id")
```

Comment:
This is a slightly weird last step, but basically we want to take all the EIA utilities that didn't get matched to SEC filers during the record linkage process, and see if we can just match them to a Ex. 21 subsidiary based on an exact match on company name. This is obviously imperfect because you can have companies that share the same name but aren't actually matches. However, we have basically no other useful shared columns between Ex. 21 subs and EIA utilities, and I believe that there are a lot of EIA utilities that are reported as Ex. 21 subs, so I think the benefits of adding in a bunch of true positive matches outweighs the cons of also adding in a bunch of false positives. I think a user is likely to want to see a possible connection to EIA and then assess for themselves whether that is a valid connection vs not having the connection at all.

Reply:
We should probably think of a way to communicate uncertainty to downstream users through documentation or something.

Reply:
One idea i had was to include the `match_probability` column that `splink` provides in the output SEC table. The threshold for matching is .95, so all probabilities would be between .95-1 which is maybe not that helpful from a "human intuition perspective". I think for now, the best thing to do is just add table level documentation that conveys  that the `utility_id_eia` column is modeled and the whole extraction of data is modeled.